runtime-rs: ch: Change state when VM stopped

jodh-intel opened the pull request:
Make the CH (Cloud Hypervisor) `stop_vm()` method check the VM state before attempting to stop the VM, and update the state once the VM has stopped.

This avoids the method failing if called multiple times which will happen if the workload exits before the container manager requests that the container stop.

This change ensures the CH driver finishes cleanly.

Fixes: #8629.

dborquez reviewed (approved):
LGTM, thanks @jodh-intel

amshinde commented on src/runtime-rs/crates/hypervisor/src/ch/inner_hypervisor.rs:
```diff
@@ -631,8 +631,20 @@ impl CloudHypervisorInner {
+        // which results in this method being called potentially a second
+        // time. Without this check, we'll return an error representing EPIPE
+        // since the CH API socket is at that point invalid.
+        if self.state != VmmState::VmRunning {
```
I was going to suggest to explicitly check for Stopped State here, but looks like there are only 3 states defined. I would eventually like to explicitly distinguish between NotReady and Stopped, but this would work for now.

amshinde commented:
/test

jodh-intel commented:
/retest-arm

jodh-intel commented:
@jongwu - Is there a known problem with the ARM jenkins CI as I'm seeing lots of errors like this:

```
host system doesn't support vsock: stat /dev/vhost-vsock: no such file or directory
```

jodh-intel commented:
> is there a reason we don't update the state within cloud_hypervisor_shutdown

@cmaf - I've moved the state handling to the Hypervisor API calls now for clarity and consistency (see new commit).

> Also I wanted to check I'm understanding this correctly, if cloud_hypervisor_shutdown returns an error the state will remain VmRunning?

@cmaf - Good catch. I've updated the branch to set the state to `NotReady` before that call as, conveniently, that's the state we want on success and failure ;)

jodh-intel commented:
/test

jodh-intel commented:
/test
